Automattic-Tracks-iOS/Crash Logging/SentryClient+Shim.m

```diff
@@ -0,0 +1,40 @@
+#import "SentryClient+Shim.h"
+
+@interface SentryClient(Internal)
+- (SentryEvent *_Nullable)prepareEvent:(SentryEvent *)event
+                             withScope:(SentryScope *)scope
+                alwaysAttachStacktrace:(BOOL)alwaysAttachStacktrace
+                          isCrashEvent:(BOOL)isCrashEvent;
+@end
+
+@implementation SentryClient(Shim)
+
+/*
+ * A wrapper around a private method in `SentryClient`. If it's unavailable (say the library changes),
+ * the worst that should happen is that the stack trace is no longer available on events sent manually using this method.
+ *
+ * We can remove this once the Sentry SDK allows for capturing events and being notified once they're delivered.
+ * ref: https://github.com/getsentry/sentry-cocoa/issues/660
+ */
+-(SentryEvent *) addStackTraceToEvent:(SentryEvent *) event forClient:(SentryClient *) client {
+
+    SEL selector = @selector(prepareEvent:withScope:alwaysAttachStacktrace:isCrashEvent:);
+
+    if(![client respondsToSelector:selector]) {
+        return event;
+    }
+
+    SentryScope *scope = [[SentrySDK currentHub] getScope];
+    SentryEvent *eventWithStackTrace = [self prepareEvent:event withScope:scope alwaysAttachStacktrace:YES isCrashEvent:NO];
+
+    if(eventWithStackTrace == nil) {
+        return event;
+    }
+
+    if(eventWithStackTrace.threads.firstObject.stacktrace != nil) {
+        eventWithStackTrace.stacktrace = eventWithStackTrace.threads.firstObject.stacktrace;
+    }
```

Comment:
Yeah, for now it's a reasonable assumption – this functionality is designed for logging crashes on startup so that we can log the message before another crash happens.

As part of the core project to re-work app initialization, I'm hoping to kill this code and be able to put up a UI for something like "the app crashed last time – would you like to try again or reset?" and in that time Sentry could do its thing and we could kill this code.

Reply:
>  put up a UI for something like "the app crashed last time – would you like to try again or reset?"

That would be such a great thing to have!